htdocs/core/triggers/interface_50_modAgenda_ActionsAuto.class.php

```diff
@@ -958,11 +960,14 @@ public function runTrigger($action, $object, User $user, Translate $langs, Conf
 		}
 		//var_dump($societeforaction);var_dump($contactforaction);var_dump($elementid);var_dump($elementtype);exit;
 
+		$type_code = '';
+		if (strpos($action, 'SENTBYMAIL')) $type_code = 'AC_EMAIL_OUT';
+
 		// Insertion action
 		require_once DOL_DOCUMENT_ROOT.'/comm/action/class/actioncomm.class.php';
 		$actioncomm = new ActionComm($this->db);
-		$actioncomm->type_code   = $object->actiontypecode; // Type of event ('AC_OTH', 'AC_OTH_AUTO', 'AC_XXX'...)
-		$actioncomm->code        = 'AC_'.$action;
+		$actioncomm->type_code   = !empty($type_code) ? $type_code : $object->actiontypecode; // Type of event ('AC_OTH', 'AC_OTH_AUTO', 'AC_XXX'...)
```

Comment:
The caller should be able to define the value by setting $object->actiontypecode
So this value must used in priority and only after type_code can be used.
Which value do you have for $object->actiontypecode when action is *SENTBYMAIL* ?